WordPress/gutenberg
# Font Library: fix infinite loop when calling wp_get_upload_dir in a function that's used to filter font_dir

## What?
Fix infinite loop when calling `wp_get_upload_dir` in a function that's used to filter `font_dir`.
An alternative approach to: https://github.com/WordPress/wordpress-develop/pull/6198

## Why?
This avoids an infinite loop that can occur if `wp_upload_dir()` is called inside a 'font_dir' callback.
Fix: https://github.com/WordPress/gutenberg/issues/58696

## Testing instructions
1. Use `wp_get_upload_dir()` inside a function used to filter `upload_dir` as in the following example:

```php
 function alter_wp_fonts_dir( $defaults ) {
	$wp_upload_dir = wp_get_upload_dir();
	$uploads_basedir = $wp_upload_dir['basedir'];
	$uploads_baseurl = $wp_upload_dir['baseurl'];

	$fonts_dir = $uploads_basedir . '/fonts';
	// Generate the URL for the fonts directory from the font dir.
	$fonts_url = str_replace( $uploads_basedir, $uploads_baseurl, $fonts_dir );

	$defaults['path'] = $fonts_dir;
	$defaults['url']  = $fonts_url;

	return $defaults;
}
add_filter( 'font_dir', 'alter_wp_fonts_dir' );
```
2. Upload fonts using the font library
3. Check that the fonts were successfully uploaded to the path set by the filter.
(`wp-content/uploads/fonts` in the example provided).
